SSR-3810: improve Table plus minus button (#1596)
kintone-labs/kintone-ui-component: docs/document/docs/components/desktop/table.md

| Name | Type | Default | Description | Remark |
| :--- | :--- | :--- | :--- | :--- |

> | actionButtonPosition | string | "right"  | Position of the fixed action button column | Available options:<br/>"right" : Right edge<br/>"left" : Left edge |

Reviewer: ```suggestion
| actionButtonPosition | string | "right"  | Position of the fixed add/remove row button column | Available options:<br/>"right" : Right edge of window<br/>"left" : Left edge of window |
```

| --kuc-table-header-height | |
| --kuc-table-header-\{index\}-width | <li>This property allows you to set the width of certain table columns based on their index values</li><li>For example, you can use `--kuc-table-header-0-width` to set the width of the first column</li><li>Note that the index values start from `0`, where `0` corresponds to the first column</li> |
| --kuc-table-header-width | <li>This property is used to set the width for all columns in a table, it defines a uniform width for every column</li><li>If you need to set a specific width for a single column, you can use `--kuc-table-header-{index}-width` property</li><li>If you want to add the element that will expand/contract as a child component, you may be able to preserve this behavior by setting `--kuc-table-header-width: auto`</li> |
| --kuc-table-action-button-background-color | |

> | --kuc-table-action-button-left | <ul><li>This property is used to set the offset of the left edge of the action button column from the left edge of Table's nearest scrolling ancestor</li><li>It works when `actionButtonPosition` is left</li><li>For example, when a child table is set up in a table and `actionButtonPosition` is left for both the child table and the parent table</li><ul><li>You can set the width of the action button column of the parent table to `--kuc-table-action-button-left` in the child table to avoid the problem that the action button columns may overwrite each other</li><li>In general, it is set to 77px if both the add/remove row buttons of the parent table are shown, or 41px if only one of the add/remove row buttons of the parent table is shown</li></ul></ul> |

Reviewer: ```suggestion
| --kuc-table-action-button-left | <ul><li>This property is used to set the offset of the left edge of the add/remove row button column from the left edge of Table's nearest scrolling ancestor</li><li>It works when `actionButtonPosition` is left</li><li>For example, when a child table is set up in a table and `actionButtonPosition` is left for both the child table and the parent table</li><ul><li>You can set the width of the add/remove row button column of the parent table to `--kuc-table-action-button-left` in the child table to avoid the problem that the add/remove row button columns may overwrite each other</li><li>In general, it is set to 77px if both the add/remove row buttons of the parent table are shown, or 41px if only one of the add/remove row buttons of the parent table is shown</li></ul></ul> |
```

> | --kuc-table-action-button-right | <ul><li>This property is used to set the offset of the right edge of the action button column from the right edge of Table's nearest scrolling ancestor</li><li>It works when `actionButtonPosition` is right</li><li>For example, when a child table is set up in a table and `actionButtonPosition` is right for both the child table and the parent table</li><ul><li>You can set the width of the action button column of the parent table to `--kuc-table-action-button-right` in the child table to avoid the problem that the action button columns may overwrite each other</li><li>In general, it is set to 77px if both the add/remove row buttons of the parent table are shown, or 41px if only one of the add/remove row buttons of the parent table is shown</li></ul></ul> |

Reviewer: What is Table's nearest scrolling ancestor?

Author: > What is Table's nearest scrolling ancestor?

When traversing the DOM tree upwards from Table, the nearest scrolling ancestor is the first element that has a "scrolling mechanism".
Ref. https://developer.mozilla.org/en-US/docs/Web/CSS/position#sticky

Reviewer: Got it. Do you think engineers can understand only with the word "nearest scrolling ancestor" (直近のスクロールする祖先)?

Is it better with "nearest ancestor element that has scrolling mechanism" (直近のスクロールの仕組みを持つ祖先要素)?

Reviewer: Thank you!

> | --kuc-table-action-button-right | <ul><li>This property is used to set the offset of the right edge of the action button column from the right edge of Table's nearest scrolling ancestor</li><li>It works when `actionButtonPosition` is right</li><li>For example, when a child table is set up in a table and `actionButtonPosition` is right for both the child table and the parent table</li><ul><li>You can set the width of the action button column of the parent table to `--kuc-table-action-button-right` in the child table to avoid the problem that the action button columns may overwrite each other</li><li>In general, it is set to 77px if both the add/remove row buttons of the parent table are shown, or 41px if only one of the add/remove row buttons of the parent table is shown</li></ul></ul> |

Reviewer: ```suggestion
| --kuc-table-action-button-right | <ul><li>This property is used to set the offset of the right edge of the add/remove row button column from the right edge of Table's nearest scrolling ancestor</li><li>It works when `actionButtonPosition` is right</li><li>For example, when a child table is set up in a table and `actionButtonPosition` is right for both the child table and the parent table</li><ul><li>You can set the width of the add/remove row button column of the parent table to `--kuc-table-action-button-right` in the child table to avoid the problem that the add/remove row button columns may overwrite each other</li><li>In general, it is set to 77px if both the add/remove row buttons of the parent table are shown, or 41px if only one of the add/remove row buttons of the parent table is shown</li></ul></ul> |
```

| --kuc-table-action-button-background-color | |
| --kuc-table-action-button-left | <ul><li>This property is used to set the offset of the left edge of the add/remove row button column from the left edge of Table's nearest scrolling ancestor</li><li>It works when `actionButtonPosition` is left</li><li>For example, when a child table is set up in a table and `actionButtonPosition` is left for both the child table and the parent table</li><ul><li>You can set the width of the add/remove row button column of the parent table to `--kuc-table-action-button-left` in the child table to avoid the problem that the add/remove row button columns may overwrite each other</li><li>In general, it is set to 77px if both the add/remove row buttons of the parent table are shown, or 41px if only one of the add/remove row buttons of the parent table is shown</li></ul></ul> |

> | --kuc-table-action-button-right | <ul><li>This property is used to set the offset of the right edge of the add/remove row button column from the right edge of Table's nearest scrolling ancestor</li><li>It works when `actionButtonPosition` is right</li><li>For example, when a child table is set up in a table and `actionButtonPosition` is right for both the child table and the parent table</li><ul><li>You can set the width of the add/remove row button column of the parent table to `--kuc-table-action-button-right` in the child table to avoid the problem that the add/remove row button columns may overwrite each other</li><li>In general, it is set to 77px if both the add/remove row buttons of the parent table are shown, or 41px if only one of the add/remove row buttons of the parent table is shown</li></ul></ul> |

Reviewer: > 41px if only one of the add/remove row buttons of the parent table is shown

User wanna display add and remove row button fully in most cases, so considering that, I think we don't need this "only one" pattern. What do you think?

Author: When there is one row, only the add button is shown, in which case it needs to be set to 41px;
When there are multiple rows, both the add/delete buttons are shown, in which case it needs to be set to 77px.

Reviewer: Ok, you assume that developer wanna change the px following the change of number of the row.
I think it's ok to keep the current description. Thanks for the explanation.
